Add fileStorageUploadField and fileUploadField Components to UI Samples

AmirShelli opened the pull request:
This pull request introduces two new components to the UI samples: `fileStorageUploadField` and `fileUploadField`.
Added their description and its Russian localization.

honest-niceman commented on src/main/java/io/jmix/uisamples/entity/Document.java:
```diff
@@ -0,0 +1,36 @@
+
+import java.util.UUID;
+
+@JmixEntity
```
Please look other entities declaration. @Entity and @Table annotations are missing here

honest-niceman commented on src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:
```diff
@@ -0,0 +1,46 @@
+public class FileStorageUploadFieldSample extends StandardView {
+    @Autowired
+    private TemporaryStorage temporaryStorage;
+    @ViewComponent
```
Do not mix @Autowired and @ViewComponent fields. Please define all autowired field first, then all @ViewComponent

honest-niceman commented on src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:
```diff
@@ -0,0 +1,46 @@
+    public void onFileStorageUploadFieldFileUploadSucceeded(
+            final FileUploadSucceededEvent<FileStorageUploadField> event) {
+        Receiver receiver = event.getReceiver();
+        if (receiver instanceof FileTemporaryStorageBuffer) {
```
Use smart instanceof construction from java17. E.g:

if (receiver instanceof FileTemporaryStorageBuffer storageBuffer) {

honest-niceman commented on src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:
```diff
@@ -0,0 +1,46 @@
+@ViewController("file-storage-upload-field")
+@ViewDescriptor("file-storage-upload-field.xml")
+public class FileStorageUploadFieldSample extends StandardView {
+    @Autowired
```
Add empty line before first field

honest-niceman commented on src/main/resources/io/jmix/uisamples/view/flowui/components/avatar/avatar-en.html:
```diff
@@ -1,4 +1,4 @@
-<p>A avatar can contain a name, an abbreviation, an icon. By default, the abbreviation is the first characters of the name.</p>
+<p>An avatar can contain a name, an abbreviation, an icon. By default, the abbreviation is the first characters of the name.</p>
```
Basically, this change has nothing related to this PR. Isn't it?

honest-niceman commented on src/main/resources/io/jmix/uisamples/menu/uisamples-menu.xml:
```diff
@@ -20,6 +20,18 @@
     <menu id="components-root" url="flow-ui/vc/components">
         <about id="components" location="components/about/about.html"/>
         <item id="avatar" page="avatar"/>
+        <menu id="upload-field">
```
The component should be added in alphabetic order, not as the second item

honest-niceman commented on src/main/resources/io/jmix/uisamples/menu/uisamples-menu.xml:
```diff
@@ -20,6 +20,18 @@
+        <menu id="upload-field">
+            <item id="file-upload-field" page="file-upload-field">
+                <otherFiles>
+                    <file name="io/jmix/uisamples/entity/Picture.java"/>
```
Check out the documentation link. It leads to the 404 page

honest-niceman commented on src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/fileuploadfield/FileUploadFieldSample.java:
```diff
@@ -0,0 +1,28 @@
+                        + fileName + "\" has been uploaded successfully.").withThemeVariant(NotificationVariant.LUMO_PRIMARY)
+                .show();
+    }
+}
```
Keep looking at file ends. Only one empty line at the end. Not two or zero empty lines.

honest-niceman commented on src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Subscribe("fileStorageUploadField")
+    public void onFileStorageUploadFieldFileUploadSucceeded(
+            final FileUploadSucceededEvent<FileStorageUploadField> event) {
```
Do we have a linter for exceeded line length? If not, i think it's better to not split method declaration into two lines

honest-niceman commented on src/main/resources/io/jmix/uisamples/messages_en.properties:
```diff
@@ -165,6 +172,10 @@ uisamples-menu-config.components=Components overview
 
 uisamples-menu-config.avatar = Avatar
 
+uisamples-menu-config.upload-field = UploadField
```
I think it's better to name it "Managing files"

honest-niceman reviewed (changes requested):
Please check out ny suggestions

AmirShelli commented:
Implemented the requested changes.

honest-niceman reviewed (approved):
Looks good to me

honest-niceman commented:
@knstvk please take a look, LGTM

knstvk commented:
@AmirShelli - please explain in the description:
- how to use the example: what file types and sizes are accepted.
- the fact that the example of fileStorageUploadField does not actually saves the file in the storage. Add a comment to the code explaining that it's just a demo code and should not be used in production.

Also, how the drop zone is supposed to work? When I drag-drop a file to the component from macOS Finder, the file is downloaded instead of uploaded.

AmirShelli commented:
@knstvk 
Implemented the requested changes:
* Adjusted the description accordingly
* Included instructions on how to use the drag-and-drop zone
* Changed the section name to "File Upload Fields"
